amd_openvx_extensions/amd_rpp/source/CropMirrorNormalize.cpp

```diff
@@ -0,0 +1,509 @@
+#elif ENABLE_HIP
+    STATUS_ERROR_CHECK(vxQueryNode(node, VX_NODE_ATTRIBUTE_AMD_HIP_STREAM, &data->handle.hipstream, sizeof(data->handle.hipstream)));
+#endif  
+    STATUS_ERROR_CHECK(vxCopyScalar((vx_scalar)parameters[11], &data->layout, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxCopyScalar((vx_scalar)parameters[12], &roiType, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxReadScalarValue((vx_scalar)parameters[13], &data->nbatchSize));
+    STATUS_ERROR_CHECK(vxCopyScalar((vx_scalar)parameters[14], &data->device_type, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+
+    if(roiType == 1)
+        data->roiType = RpptRoiType::XYWH;
+    else
+        data->roiType = RpptRoiType::LTRB;
+
+    // Querying for input tensor
+    data->src_desc_ptr = &data->srcDesc;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &data->src_desc_ptr->numDims, sizeof(data->src_desc_ptr->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, &data->in_tensor_dims, sizeof(vx_size) * data->src_desc_ptr->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DATA_TYPE, &data->in_tensor_type, sizeof(data->in_tensor_type)));
+    if(data->in_tensor_type == vx_type_e::VX_TYPE_UINT8)
+    {
+        data->src_desc_ptr->dataType = RpptDataType::U8;
+    }
+    else if (data->in_tensor_type == vx_type_e::VX_TYPE_FLOAT32)
+    {
+        data->src_desc_ptr->dataType = RpptDataType::F32;
+    }
+    // VX_TYPE_FLOAT16 is not supported. Have to enable it once support is added.
+    // else if (data->in_tensor_type == vx_type_e::VX_TYPE_FLOAT16)
+    // {
+    //     data->src_desc_ptr->dataType = RpptDataType::F16;
+    // }
+    data->src_desc_ptr->offsetInBytes = 0;
+
+    // Querying for output tensor
+    data->dst_desc_ptr = &data->dstDesc;
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_NUMBER_OF_DIMS, &data->dst_desc_ptr->numDims, sizeof(data->dst_desc_ptr->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DIMS, &data->out_tensor_dims, sizeof(vx_size) * data->dst_desc_ptr->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2],VX_TENSOR_DATA_TYPE, &data->out_tensor_type, sizeof(data->out_tensor_type)));
+    if(data->out_tensor_type == vx_type_e::VX_TYPE_UINT8)
+    {
+        data->dst_desc_ptr->dataType= RpptDataType::U8;
+    }
+    else if (data->out_tensor_type == vx_type_e::VX_TYPE_FLOAT32)
+    {
+        data->dst_desc_ptr->dataType = RpptDataType::F32;
+    }
+    // VX_TYPE_FLOAT16 is not supported. Have to enable it once support is added.
+    // else if (data->out_tensor_type == vx_type_e::VX_TYPE_FLOAT16)
+    // {
+    //     data->src_desc_ptr->dataType = RpptDataType::F16;
+    // }
+    data->dst_desc_ptr->offsetInBytes = 0;
+
+    if(data->layout == 0) // NHWC
```

Comment:
please use enum for Layout instead of numeric value

Reply:
We will not be able to pass a rocAL enum to the amd_rpp files, as the parameters can only be vx_scalar types

Reply:
I think you can still pass a scalar as VX_ENUM type. Here in this line you can check the type with any declared enum type. Please make this change in the next PR